doc/user/content/overview/architecture.md

```diff
@@ -1,31 +1,99 @@
+
+The key concepts and nouns in storage that users have to interact with are `CONNECTION`s, `SOURCE`s, and `SINK`s.
+
+### Connections
+
+A connection describes how to connect and authenticate to an external system that Materialize will read data from or write data to. A connection is a logical entity. Materialize will actually connect to the external system specified in a connection when a source or sink is created using that connection.
+
+### Sources
+
+A source is the SQL respresentation of an input data collection from an external system (e.g. a Kafka topic). A source describes a location within an external system (e.g. a Kafka topic on a specified Kafka cluster) that Materialize will read data from, and provides details on how to decode and interpret the incoming data from that particular location. Materialize will start reading and maintaining a stored collection (an explicitly timestamped history of changes to the input data) as soon as the source is created.
+
+The computational resources available to decode and interpret the input data, and maintain the stored collection can be scaled up or down by modifying the source's size with `ALTER SOURCE`.
+
+### Sinks
+
+Sinks are the inverse of sources. A sink describes a location within an external system (e.g. a Kafka topic on a specified Kafka cluster) that Materialize will write data to and provides details on how to encode the written data. Sinks can be defined over existing sources, materialized views, or tables. When a sink is defined, Materialize automatically generates the required schema, and writes the stream of changes to the specified source, materialized view, or table in the specified format.
+
+The computational resources available to read from persistent storage and encode data before writing it out to the external system can be scaled up or down by modifying the sink's size with `ALTER SINK`.
+
+## Compute
+
+The compute layer is responsible for efficiently computing and maintaining views over data collections maintained by the Storage layer.
+
+The key concepts and nouns that users have to interact with are `VIEW`s, `CLUSTER`s, `CLUSTER REPLICA`s, `INDEX`es, and `MATERIALIZED VIEW`s.
+
+### (Unmaterialized) Views
 
-The architecture is broken into three logical components:
+Unmaterialized views represent a query that a user can save and associate with a name. These are used primarily as shorthand for some lengthy, complicated `SELECT` statement. Materialize will not compute or maintain an unmaterialized view when it is created. Materialize will compute the view (i.e. compute the underlying `SELECT` statement) if the view is `SELECT`ed.
 
-### Storage
+Repeatedly querying an unmaterialized view without any indexes is discouraged because Materialize will repeatedly compute and then discardiscard the output collections. It is recommended in that case to either create indexes based on the repeated query or a materialized view instead.
```

Comment:
```suggestion
Repeatedly querying an unmaterialized view without any indexes is discouraged because Materialize will repeatedly compute and then discard the output collections. It is recommended in that case to either create indexes based on the repeated query or a materialized view instead.
```